build.rs

```diff
@@ -29,5 +30,69 @@ fn main() -> Result<(), Box<dyn Error>> {
     writeln!(f, "pub const USER_AGENT: &str = \"sentry-cli/{arch}\";")?;
     println!("cargo:rerun-if-changed=build.rs\n");
 
+    if platform == "darwin" {
+        println!("cargo:rerun-if-changed=native/swift/AssetCatalogParser");
+
+        env::set_current_dir("native/swift/AssetCatalogParser")
+            .expect("Failed to change to AssetCatalogParser directory");
+
+        let target_dir = "../../../target/swift-bridge";
+
+        let _ = std::fs::remove_dir_all(".build");
+        let _ = std::fs::remove_file(format!("{}/libswiftbridge.a", target_dir));
+
+        let status = Command::new("swift")
+            .args([
+                "build",
+                "-c",
+                "release",
+                "--triple",
+                &format!("{}-apple-macosx11", arch),
+            ])
+            .status()
+            .expect("Failed to compile SPM");
+
+        assert!(status.success(), "swift build failed");
+
+        let target_dir = "../../../target/swift-bridge";
+
+        std::fs::create_dir_all(target_dir)
+            .expect("Failed to create target/swift-bridge directory");
+
+        let status = Command::new("ar")
+            .args([
+                "crus",
+                &format!("{}/libswiftbridge.a", target_dir),
+                ".build/release/AssetCatalogParser.build/AssetCatalogReader.swift.o",
+                ".build/release/ObjcSupport.build/safeValueForKey.m.o",
+            ])
+            .status()
+            .expect("Failed to create static library");
+
+        assert!(status.success(), "ar failed");
+
+        env::set_current_dir("../../../").expect("Failed to change back to original directory");
+
+        println!("cargo:rustc-link-search=native=target/swift-bridge");
+        println!("cargo:rustc-link-lib=static=swiftbridge");
+
+        println!("cargo:rustc-link-arg=-F");
+        println!("cargo:rustc-link-arg=/System/Library/PrivateFrameworks");
+        println!("cargo:rustc-link-lib=framework=CoreUI");
```

Comment:
[question] From what I can tell, these libraries will be dynamically linked.

I'm not super familiar with the Apple ecosystem, but from the name of this path `PrivateFrameworks`, I am guessing these might be private macOS APIs, which are subject to change? Couldn't this potentially make Sentry CLI break with older or newer macOS versions where these system libraries are changed? If so, is there any way we can avoid this?